app/controllers/submissions_controller.rb

```diff
@@ -285,6 +284,15 @@ def update_files
       render :file_manager, id: assignment_id
       return
     end
+    if (params[:new_files]!=nil)
```

Comment:
Explicit non-nil checks are usually redundant.<br>Surrounding space missing for operator '!='.